cmd/juju/application/removeapplication.go

```diff
@@ -239,8 +239,13 @@ func (c *removeApplicationCommand) removeApplications(
 	for i, name := range c.ApplicationNames {
 		result := results[i]
 		if result.Error != nil {
-			ctx.Infof("removing application %s failed: %s", name, result.Error)
 			anyFailed = true
+			err := result.Error.Error()
+			if result.Error.Code == params.CodeNotSupported {
+				ctx.Warningf("changes detected to application %s", name)
+				err = errors.New("retry once any changes have completed").Error()
```

Comment:
This results in an error message like:

```
removing application do-not-remove failed: 
retry once any changes have completed
```

There's not enough context there for the user. What changes they will ask.
How about:

```
removing application do-not-remove failed: 
another user was updating application; please try again
````

Reply:
Sounds good